File: frontend/capture_screenshots.py
"""
Script para capturar screenshots do Prova AI para o tutorial
Execute com: playwright install chromium && python capture_screenshots.py
"""

import os
from pathlib import Path
from playwright.sync_api import sync_playwright

BASE_URL = "https://ia-educacao-v2.onrender.com"
OUTPUT_DIR = Path(__file__).parent / "tutorial-images"

# Garantir que a pasta existe
OUTPUT_DIR.mkdir(parents=True, exist_ok=True)

def capture_screenshots():
    print("🚀 Iniciando captura de screenshots...\n")
    
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        context = browser.new_context(
            viewport={"width": 1400, "height": 900},
            device_scale_factor=2  # Retina quality
        )
        page = context.new_page()
        
        try:
            # 1. Dashboard Principal
            print("📸 1/8 - Dashboard...")
            page.goto(BASE_URL, wait_until="networkidle")
            page.wait_for_timeout(3000)
            page.screenshot(path=str(OUTPUT_DIR / "01-dashboard.png"))
            
            # 2. Sidebar expandido
            print("📸 2/8 - Sidebar navegação...")
            tree_item = page.query_selector(".tree-item")
            if tree_item:
                tree_item.click()
                page.wait_for_timeout(500)
            page.screenshot(
                path=str(OUTPUT_DIR / "02-sidebar.png"),
                clip={"x": 0, "y": 0, "width": 320, "height": 900}
            )
            
            # 3. Chat com IA
            print("📸 3/8 - Chat com IA...")
            chat_link = page.query_selector("text=Chat com IA")
            if chat_link:
                chat_link.click()
                page.wait_for_timeout(2500)
            page.screenshot(path=str(OUTPUT_DIR / "03-chat.png"))
            
            # 4. Chat - Modo filtrar
            print("📸 4/8 - Chat com filtros...")
            filter_btn = page.query_selector('[data-mode="filtrar"]') or page.query_selector("text=Filtrar")
            if filter_btn:
                filter_btn.click()
                page.wait_for_timeout(1000)
            page.screenshot(path=str(OUTPUT_DIR / "04-chat-filtros.png"))
            
            # 5. Modal de Configurações
            print("📸 5/8 - Configurações IA...")
            config_btn = page.query_selector("text=Configurações IA")
            if config_btn:
                config_btn.click()
                page.wait_for_timeout(1500)
            page.screenshot(path=str(OUTPUT_DIR / "05-config-apikeys.png"))
            
            # 6. Aba de Modelos (dentro do modal de settings)
            print("📸 6/8 - Configuração de modelos...")
            # Clicar na aba dentro do modal
            modelos_tab = page.query_selector('#modal-settings [data-tab="modelos"]')
            if modelos_tab:
                modelos_tab.click()
                page.wait_for_timeout(1000)
            page.screenshot(path=str(OUTPUT_DIR / "06-config-modelos.png"))
            
            # Fechar modal clicando no X dentro do modal
            close_btn = page.query_selector("#modal-settings .modal-close")
            if close_btn:
                close_btn.click()
                page.wait_for_timeout(500)
            else:
                page.keyboard.press("Escape")
                page.wait_for_timeout(500)
            
            # 7. Modal Nova Matéria
            print("📸 7/8 - Nova Matéria...")
            # Aguardar modal fechar
            page.wait_for_timeout(500)
            nova_materia_btn = page.query_selector('button:has-text("Nova Matéria")')
            if nova_materia_btn:
                nova_materia_btn.click()
                page.wait_for_timeout(1000)
            page.screenshot(path=str(OUTPUT_DIR / "07-nova-materia.png"))
            
            # Fechar modal
            page.keyboard.press("Escape")
            page.wait_for_timeout(500)
            
            # 8. Navegar para matéria/atividade
            print("📸 8/8 - Tela de matéria...")
            inicio_link = page.query_selector('a:has-text("Início")')
            if inicio_link:
                inicio_link.click()
                page.wait_for_timeout(1000)
            
            # Clicar em uma matéria se existir
            materia_card = page.query_selector(".materia-card")
            if materia_card:
                materia_card.click()
                page.wait_for_timeout(1500)
            page.screenshot(path=str(OUTPUT_DIR / "08-materia-view.png"))
            
            print("\n✅ Screenshots capturados com sucesso!")
            print(f"📁 Salvos em: {OUTPUT_DIR}")
            
            # Listar arquivos criados
            files = list(OUTPUT_DIR.glob("*.png"))
            print(f"\n📷 {len(files)} imagens capturadas:")
            for f in sorted(files):
                print(f"   - {f.name}")
                
        except Exception as e:
            print(f"❌ Erro durante captura: {e}")
        finally:
            browser.close()

if __name__ == "__main__":
    capture_screenshots()

File: frontend/capture-screenshots.js
/**
 * Script para capturar screenshots do Prova AI para o tutorial
 * Execute com: npx playwright install chromium && node capture-screenshots.js
 */

const { chromium } = require('playwright');
const fs = require('fs');
const path = require('path');

const BASE_URL = 'https://ia-educacao-v2.onrender.com';
const OUTPUT_DIR = path.join(__dirname, 'tutorial-images');

// Garantir que a pasta existe
if (!fs.existsSync(OUTPUT_DIR)) {
    fs.mkdirSync(OUTPUT_DIR, { recursive: true });
}

async function captureScreenshots() {
    console.log('🚀 Iniciando captura de screenshots...\n');
    
    const browser = await chromium.launch({ 
        headless: true,
        args: ['--no-sandbox']
    });
    
    const context = await browser.newContext({
        viewport: { width: 1400, height: 900 },
        deviceScaleFactor: 2, // Retina quality
    });
    
    const page = await context.newPage();
    
    try {
        // 1. Dashboard Principal
        console.log('📸 1/8 - Dashboard...');
        await page.goto(BASE_URL, { waitUntil: 'networkidle' });
        await page.waitForTimeout(2000);
        await page.screenshot({ 
            path: path.join(OUTPUT_DIR, '01-dashboard.png'),
            fullPage: false 
        });
        
        // 2. Sidebar expandido (hover na matéria se existir)
        console.log('📸 2/8 - Sidebar navegação...');
        // Tentar expandir uma matéria na sidebar
        const materiaItem = await page.$('.tree-item');
        if (materiaItem) {
            await materiaItem.click();
            await page.waitForTimeout(500);
        }
        await page.screenshot({ 
            path: path.join(OUTPUT_DIR, '02-sidebar.png'),
            clip: { x: 0, y: 0, width: 320, height: 900 }
        });
        
        // 3. Chat com IA
        console.log('📸 3/8 - Chat com IA...');
        await page.click('text=Chat com IA');
        await page.waitForTimeout(2000);
        await page.screenshot({ 
            path: path.join(OUTPUT_DIR, '03-chat.png'),
            fullPage: false 
        });
        
        // 4. Chat - Painel de filtros (se existir botão de filtro)
        console.log('📸 4/8 - Chat com filtros...');
        const filterBtn = await page.$('[data-mode="filtrar"]') || await page.$('text=Filtrar');
        if (filterBtn) {
            await filterBtn.click();
            await page.waitForTimeout(1000);
        }
        await page.screenshot({ 
            path: path.join(OUTPUT_DIR, '04-chat-filtros.png'),
            fullPage: false 
        });
        
        // 5. Modal de Configurações (API Keys)
        console.log('📸 5/8 - Configurações IA...');
        await page.click('text=Configurações IA');
        await page.waitForTimeout(1500);
        await page.screenshot({ 
            path: path.join(OUTPUT_DIR, '05-config-apikeys.png'),
            fullPage: false 
        });
        
        // 6. Aba de Modelos no modal de config
        console.log('📸 6/8 - Configuração de modelos...');
        const modelosTab = await page.$('text=Modelos') || await page.$('[data-tab="modelos"]');
        if (modelosTab) {
            await modelosTab.click();
            await page.waitForTimeout(1000);
        }
        await page.screenshot({ 
            path: path.join(OUTPUT_DIR, '06-config-modelos.png'),
            fullPage: false 
        });
        
        // Fechar modal
        const closeBtn = await page.$('.modal-close') || await page.$('button:has-text("×")');
        if (closeBtn) {
            await closeBtn.click();
            await page.waitForTimeout(500);
        }
        
        // 7. Modal Nova Matéria
        console.log('📸 7/8 - Nova Matéria...');
        await page.click('text=Nova Matéria');
        await page.waitForTimeout(1000);
        await page.screenshot({ 
            path: path.join(OUTPUT_DIR, '07-nova-materia.png'),
            fullPage: false 
        });
        
        // Fechar modal
        await page.keyboard.press('Escape');
        await page.waitForTimeout(500);
        
        // 8. Navegar para uma atividade (se existir)
        console.log('📸 8/8 - Tela de atividade...');
        await page.click('text=Início');
        await page.waitForTimeout(1000);
        
        // Tentar clicar em uma matéria existente
        const materiaCard = await page.$('.materia-card') || await page.$('[onclick*="viewMateria"]');
        if (materiaCard) {
            await materiaCard.click();
            await page.waitForTimeout(1500);
            await page.screenshot({ 
                path: path.join(OUTPUT_DIR, '08-materia-view.png'),
                fullPage: false 
            });
        } else {
            // Screenshot do dashboard mesmo
            await page.screenshot({ 
                path: path.join(OUTPUT_DIR, '08-materia-view.png'),
                fullPage: false 
            });
        }
        
        console.log('\n✅ Screenshots capturados com sucesso!');
        console.log(`📁 Salvos em: ${OUTPUT_DIR}`);
        
    } catch (error) {
        console.error('❌ Erro durante captura:', error.message);
    } finally {
        await browser.close();
    }
}

captureScreenshots();
